src/components/ZenodoImporter.vue

```diff
@@ -0,0 +1,298 @@
+      this.error = "";
+    }
+  }
+
+  async importSelectedDataset(): Promise<void> {
+    if (!this.canImport || !this.path || !this.selectedDataset) return;
+
+    this.importing = true;
+    this.error = "";
+    this.importProgress = 0;
+
+    try {
+      // Get only the image files from Zenodo
+      const imageFiles = this.filteredFiles;
+      const downloadedImageFiles: File[] = [];
+
+      const nonImageFiles = this.selectedDataset.files.filter(
+        (file) => !this.filteredFiles.includes(file),
+      );
+
+      const downloadedNonImageFiles: File[] = [];
+
+      // Download each file from Zenodo
+      for (let i = 0; i < imageFiles.length; i++) {
+        const file = imageFiles[i];
+        this.currentFile = `Downloading ${file.key}...`;
+
+        // Set initial progress for this file
+        this.importProgress = (i / imageFiles.length) * 100;
+
+        const fileIndex = i; // Capture the current index in a closure
+
+        // Download the file with progress updates
+        const blob = await this.zenodoApi.downloadFile(
+          file.links.self,
+          (loaded, total) => {
+            // Calculate overall progress:
+            // - Completed files: (fileIndex / totalFiles) * 100
+            // - Current file: (loaded / total) / totalFiles * 100
+            const completedProgress = (fileIndex / imageFiles.length) * 100;
+            const currentFileProgress =
+              (loaded / total / imageFiles.length) * 100;
+            this.importProgress = completedProgress + currentFileProgress;
+
+            // Update status message
+            this.currentFile = `Downloading ${file.key}... (${Math.round(loaded / 1024 / 1024)}MB / ${Math.round(total / 1024 / 1024)}MB)`;
+          },
+        );
+
+        const downloadedImageFile = new File([blob], file.key);
+        downloadedImageFiles.push(downloadedImageFile);
+      }
+
+      // Download non-image files
+      for (let i = 0; i < nonImageFiles.length; i++) {
+        const file = nonImageFiles[i];
+        this.currentFile = `Downloading ${file.key}...`;
+
+        this.importProgress = (i / nonImageFiles.length) * 100;
+
+        const fileIndex = i; // Capture the current index in a closure
+
+        const blob = await this.zenodoApi.downloadFile(
+          file.links.self,
+          (loaded, total) => {
+            // Calculate overall progress:
+            // - Completed files: (fileIndex / totalFiles) * 100
+            // - Current file: (loaded / total) / totalFiles * 100
+            const completedProgress = (fileIndex / nonImageFiles.length) * 100;
+            const currentFileProgress =
+              (loaded / total / nonImageFiles.length) * 100;
+            this.importProgress = completedProgress + currentFileProgress;
+
+            // Update status message
+            this.currentFile = `Downloading ${file.key}... (${Math.round(loaded / 1024 / 1024)}MB / ${Math.round(total / 1024 / 1024)}MB)`;
+          },
```

Comment:
You probably can factorize these 2 for loops by creating a downloadFile function